server/routes/positions.js

```diff
@@ -19,4 +19,59 @@ router.get('/', function(req, res, next) {
     .catch(err => { return next(err) });
 });
 
+// Add a single position
+router.post('/', function(req, res, next) {
+  if (!req.query.role) {
+    util.throwError(400, 'Position role must not be null');
+  }
+  
+  knex('position').insert({ role: req.query.role })
+    .then(result => {
+      res.send(util.message('Successfully inserted new position: ' + req.query.role));
+    })
+    .catch(err => { return next(err) });
+});
+
+// Update a single position
+router.put('/:position_id', function(req, res, next) {
+  let errormsg = 'The following fields must not be null: ';
```

Comment:
This is nifty but you might just want to return error message in order of precedence. Otherwise can get complicated when there are many required params.

So instead of appending to an error message, if (!req.params.position_id) throwError for No Position ID found, then if (!req.body.role) throwError for no Role set, etc.